codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -925,5 +920,306 @@ final void incrementStreamableBytesForTree(int numBytes) {
+            state.enqueueFrame(frame);
+        }
+    }
+
+    /**
+     * Provides no notification or tracking of writablity changes.
+     */
+    private final class DefaultWritabilityMonitor extends AbstractWritabilityMonitor {
+        protected final Http2StreamVisitor writeAllocatedBytesVisitor = new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) {
+                state(stream).writeAllocatedBytes();
+                return true;
+            }
+        };
+
+        @Override
+        public void writePendingBytes() throws Http2Exception {
+            writePendingBytes(writeAllocatedBytesVisitor);
+        }
+
+        @Override
+        public void initialWindowSize(int newWindowSize) throws Http2Exception {
+            initialWindowSize(newWindowSize, writeAllocatedBytesVisitor);
+        }
+
+        @Override
+        public void pendingBytesChanged(int delta) {
+        }
+
+        @Override
+        public boolean isWritable(AbstractState state) {
+            return true;
+        }
+
+        @Override
+        public Listener listener() {
+            return null;
+        }
+
+        @Override
+        public void stateCancelled(AbstractState state) {
+        }
+
+        @Override
+        public void channelWritabilityChange() {
+        }
+    }
+
+    /**
+     * Writability of a {@code stream} is calculated using the following:
+     * <pre>
+     * Connection Window - Total Queued Bytes > 0 &&
+     * Stream Window - Bytes Queued for Stream > 0 &&
+     * isChannelWritable()
+     * </pre>
+     */
+    private final class ListenerWritabilityMonitor extends AbstractWritabilityMonitor {
+        private long totalPendingBytes;
+        private final Listener listener;
+        private final Http2StreamVisitor checkStreamWritabilityVisitor = new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) throws Http2Exception {
+                AbstractState state = state(stream);
+                if (isWritable(state) != state.markWritabilty()) {
+                    notifyWritabilityChanged(state);
+                }
+                return true;
+            }
+        };
+        private final Http2StreamVisitor initialWindowSizeVisitor = new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) {
+                AbstractState state = state(stream);
+                writeAllocatedBytes(state);
+                if (isWritable(state) != state.markWritabilty()) {
+                    notifyWritabilityChanged(state);
+                }
+                return true;
+            }
+        };
+        private final Http2StreamVisitor writeAllocatedBytesVisitor = new Http2StreamVisitor() {
+            @Override
+            public boolean visit(Http2Stream stream) {
+                writeAllocatedBytes(state(stream));
```

Comment:
I couldn't think of a situation where the writability would change due to a write operation (because we have already accounted for the pending bytes, and in a write pending bytes just decrement the window)